llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16142,11 +16142,8 @@ static bool atomicIgnoresDenormalModeOrFPModeIsFTZ(const AtomicRMWInst *RMW) {
 
 static OptimizationRemark emitAtomicRMWLegalRemark(const AtomicRMWInst *RMW) {
   LLVMContext &Ctx = RMW->getContext();
-  SmallVector<StringRef> SSNs;
-  Ctx.getSyncScopeNames(SSNs);
-  StringRef MemScope = SSNs[RMW->getSyncScopeID()].empty()
-                           ? "system"
-                           : SSNs[RMW->getSyncScopeID()];
+  StringRef SS = Ctx.getSyncScopeName(RMW->getSyncScopeID()).value_or("");
+  StringRef MemScope = SS.empty() ? StringRef("system") : SS;
```

Comment:
```suggestion
  StringRef MemScopeName = Ctx.getSyncScopeName(RMW->getSyncScopeID()).value_or("system");
```

Reply:
So if the value is `""` (the SyncScope string is empty), you want to keep it as `""` and not map that to `"system"`?

Reply:
Oh, no. this was supposed to map the system cases to be printed as system

Reply:
Right, so this suggestion won't do. 
The current code fills in out of bounds `SyncScope::ID` as `""` (instead of handling that as an error), and then maps `""`  to `"system"` (independently of whether `""` is from out of bounds or from system scope in the IR).